packages/jest-runtime/src/index.ts

```diff
@@ -703,10 +718,12 @@ export default class Runtime {
     options?: InternalModuleOptions,
     isRequireActual = false,
   ): T {
+    const isInternal = options?.isInternalModule ?? false;
     const moduleID = this._resolver.getModuleID(
       this._virtualMocks,
       from,
       moduleName,
+      isInternal ? undefined : {conditions: cjsConditions},
```

Comment:
this is wrong and just to avoid the fact `pathFilter` doesn't really work for what we want (it asks for every relative import _within_ a package as well). Should probably rework the test, but since this is internal loading it doesn't really matter up until we actually support `conditions` out of the box.

I do think we shouldn't use use-provided resolvers when loading our own stuff, but it is what it is